TPP.Core/Modes/Matchmode.cs

```diff
@@ -137,5 +143,34 @@ public void Dispose()
         {
             _modeBase.Dispose();
         }
+
+        public async void ResetBalances()
+        {
+            _logger.LogInformation("Resetting Balances");
+            long minimumPokeyen = _matchmodeConfig.MinimumPokeyen;
+            long subscriberMinimumPokeyen = _matchmodeConfig.SubscriberMinimumPokeyen;
+
+            List<User> poorUsers = await _userRepo.ListAllUnderPokeyen(Math.Max(minimumPokeyen, subscriberMinimumPokeyen));
+            foreach (User u in poorUsers)
+            {
+                long pokeyen = await _pokeyenBank.GetAvailableMoney(u);
+                if(u.IsSubscribed && pokeyen < subscriberMinimumPokeyen)
+                {
+                    long amountToGive = subscriberMinimumPokeyen - pokeyen;
+                    await _pokeyenBank.PerformTransaction(new Transaction<User>(u, amountToGive, TPP.Persistence.Repos.TransactionType.Welfare));
+                    // TODO whisper users informing them they have been given money
+                    _logger.LogInformation(String.Format("Subscriber {0} had their balance reset to P{1} (+P{2})", u.SimpleName, subscriberMinimumPokeyen, amountToGive));
+
+                }
+                else if (!u.IsSubscribed && u.Pokeyen < minimumPokeyen)
```

Comment:
You're using `u.Pokeyen` here, but the value returned by `await _pokeyenBank.GetAvailableMoney(u)` above. You don't need to use the available money, just the pokeyen amount on the user object should be fine, because
a) nothing except betting even reserves pokeyen (at this point in time)
b) if something were to exist to reserve pokeyen, it shouldn't cause you to get more welfare before a match